x-pack/plugins/security_solution/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -135,18 +136,44 @@ export const signalRulesAlertType = ({
       logger.debug(buildRuleMessage(`interval: ${interval}`));
       await ruleStatusService.goingToRun();
 
-      const gap = getGapBetweenRuns({ previousStartedAt, interval, from, to });
+      let gap;
+      try {
+        gap = getGapBetweenRuns({ previousStartedAt, interval, from, to });
+      } catch (exc) {
+        const errorMessage = buildRuleMessage(`${exc.message} Check logs for further details.`);
+        logger.error(errorMessage);
+        await ruleStatusService.error(errorMessage, {
+          bulkCreateTimeDurations: result.bulkCreateTimes,
+          searchAfterTimeDurations: result.searchAfterTimes,
+          lastLookBackDate: result.lastLookBackDate?.toISOString(),
+        });
+        // throw the error because we can't query later on.
+        throw Error(exc.message);
+      }
       if (gap != null && gap.asMilliseconds() > 0) {
         const fromUnit = from[from.length - 1];
-        const { ratio } = getGapMaxCatchupRatio({
-          logger,
-          buildRuleMessage,
-          previousStartedAt,
-          ruleParamsFrom: from,
-          interval,
-          unit: fromUnit,
-        });
-        if (ratio && ratio >= MAX_RULE_GAP_RATIO) {
+        let vals;
+        try {
+          vals = getGapMaxCatchupRatio({
+            logger,
+            buildRuleMessage,
+            previousStartedAt,
+            ruleParamsFrom: from,
+            interval,
+            unit: fromUnit,
+          });
```

Comment:
`getGapMaxCatchupRatio` can throw an error if "from" is NaN. If that is the case, we catch the error and log a failure for the rule status and eject early from the executor with a throw (see line 174). Again we could replace the throw with a return if there is a preference for that.